openmetadata-ui/src/main/resources/ui/src/context/PermissionProvider/PermissionProvider.test.tsx

```diff
@@ -52,7 +52,24 @@ jest.mock('../../hooks/useApplicationStore', () => {
   };
 });
 
+jest.mock('../../components/common/Loader/Loader', () => {
+  return jest.fn().mockImplementation(() => <p>Loader</p>);
+});
+
 describe('PermissionProvider', () => {
+  it('Should render loader and call getLoggedInUserPermissions', async () => {
+    render(
+      <PermissionProvider>
+        <div data-testid="children">Children</div>
+      </PermissionProvider>
+    );
+
+    // Verify that the API methods were called
+    expect(getLoggedInUserPermissions).toHaveBeenCalled();
+
+    expect(screen.getByText('Loader')).toBeInTheDocument();
```

Comment:
@Ashish8689, can we add a test for the loader should not be visible if current user is empty and api call is not being made?

Reply:
added the test!